ceci/pipeline.py

```diff
@@ -82,23 +301,183 @@ def __init__(self, stages, launcher_config):
+                      "log_dir": pipe_config["log_dir"],
+                      "resume": pipe_config["resume"]}
+
+        launcher_dict = dict(cwl=CWLPipeline,
+                             parsl=ParslPipeline,
+                             mini=MiniPipeline)
+
+        if pipe_config.get('dry_run', False):
+            pipeline_class = DryRunPipeline
+        try:
+            pipeline_class = launcher_dict[launcher_name]
+        except KeyError as msg:  #pragma: no cover
+            raise KeyError(f"Unknown pipeline launcher {launcher_name}, options are {list(launcher_dict.keys())}") from msg
+
+        p = pipeline_class(stages, launcher_config, overall_inputs=inputs, modules=modules)
+        p.load_configs(inputs, run_config, stages_config)
+        return p
+
+    @staticmethod
+    def interactive():
+        """Build and return a pipeline specifically intended for interactive use"""
+        launcher_config = dict(name="mini")
+        return MiniPipeline([], launcher_config)
+
+
+    @staticmethod
+    def build_config(pipeline_config_filename, extra_config=None, dry_run=False):
+        """Build a configuration dictionary from a yaml file and extra optional parameters
+
+        Parameters
+        ----------
+        pipeline_config_filename : str
+            The path to the input yaml file
+        extra_config : str
+            A string with extra parameters in key=value pairs
+        dry_run : bool
+            A specfic flag to build a pipeline only from dry-runs
+
+        Returns
+        -------
+        pipe_config : dict
+            The resulting configuration
+        """
+
+        # YAML input file.
+        # Load the text and then expand any environment variables
+        with open(pipeline_config_filename) as config_file:
+            raw_config_text = config_file.read()
+        config_text = os.path.expandvars(raw_config_text)
+        # Then parse with YAML
+        pipe_config = yaml.safe_load(config_text)
+
+        if extra_config:
+            override_config(pipe_config, extra_config)
+
+        # parsl execution/launcher configuration information
+        launcher_config = pipe_config.setdefault("launcher", {"name": "mini"})
+
+        # Launchers may need to know if this is a dry-run
+        launcher_config["dry_run"] = dry_run
+        pipe_config["dry_run"] = dry_run
+        return pipe_config
+
+    def __getattr__(self, name):
+        """Get a particular stage by name"""
+        try:
+            return self.stage_execution_config[name].stage_obj
+        except Exception as msg:  #pragma: no cover
+            raise AttributeError(f"Pipeline does not have stage {name}") from msg
+
+    def print_stages(self, stream=sys.stdout):
+        """Print the list of stages in this pipeline to a stream"""
+        for stage in self.stages:
+            stream.write(f"{stage.instance_name:20}: {str(stage)}")
+            stream.write("\n")
+
+    @staticmethod
```

Comment:
Hmm.   Making it a classmethod would allow you to override the (renamed) `build_config` or `create` methods, so why not.

Reply:
The reason I had it as a static was because I was thinking of it as a top-level type-agnostic entry point, but we can do that with a classmethod as well I think.